version: '3.8'

networks: 
  {{ tools.docker.network.name }}:
    external: true
    driver: bridge

services:
  uptime-kuma:
    image: {{ tools.uptime_kuma.docker_image }}:{{ tools.uptime_kuma.docker_image_tag }}
    container_name: {{ tools.uptime_kuma.container_name }}
    restart: unless-stopped
    networks:
      {{ tools.docker.network.name }}: 
        ipv4_address: {{ tools.docker.network.containers.uptime_kuma }}
    ports:
      {{ tools.uptime_kuma.ports }}
    volumes:
      - uptime_kuma:/app/data
      - /var/run/docker.sock:/var/run/docker.sock
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:3001"]
      interval: 30s
      timeout: 10s
      retries: 3
      start_period: 10s
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.kuma.rule=Host(`kuma.gorillabay.click`)"
      - "traefik.http.routers.kuma.entrypoints=websecure"
      - "traefik.http.routers.kuma.tls.certresolver=production"

volumes:
  uptime_kuma:
    name: {{ tools.uptime_kuma.docker_volume_name }}
